docs/content/guides/dagster-pipes/subprocess/reference.mdx

```diff
@@ -0,0 +1,242 @@
+    # get the Dagster Pipes context
+    context = PipesContext.get()
+    # get all extras provided by Dagster asset
+    print(context.extras)
+    # get the value of an extra
+    print(context.get_extra("foo"))
+    # get env var
+    print(os.environ["MY_ENV_VAR_IN_SUBPROCESS"])
+
+
+if __name__ == "__main__":
+    # connect to Dagster Pipes
+    with open_dagster_pipes():
+        main()
+```
+
+</TabItem>
+</TabGroup>
+
+## Working with @asset_check
+
+Sometimes, you may not want to materialize an asset, but instead want to report a data quality check result. When your asset has data quality checks defined in `@asset_check`:
+
+<TabGroup>
+<TabItem name="Define in the Dagster code">
+
+On the Dagster's side, the `PipesClientCompletedInvocation` object returned from `PipesSubprocessClient` includes a `get_asset_check_result` method, which you can use to access the <PyObject object="AssetCheckResult" /> event reported by the subprocess.
+
+```python file=/guides/dagster/dagster_pipes/subprocess/asset_check_dagster_code.py
+import shutil
+
+from dagster import (
+    AssetExecutionContext,
+    Definitions,
+    MaterializeResult,
+    PipesSubprocessClient,
+    asset,
+    asset_check,
+    file_relative_path,
+)
+from dagster._core.definitions.asset_check_result import AssetCheckResult
+
+
+@asset
+def my_asset():
+    ...
+
+
+@asset_check(asset="my_asset")
+def no_empty_order_check(
+    context, pipes_subprocess_client: PipesSubprocessClient
+) -> AssetCheckResult:
+    cmd = [
+        shutil.which("python"),
+        file_relative_path(__file__, "asset_check_external_code.py"),
+    ]
+    return pipes_subprocess_client.run(
+        command=cmd, context=context
+    ).get_asset_check_result()
+
+
+defs = Definitions(
+    assets=[my_asset],
+    asset_checks=[no_empty_order_check],
+    resources={"pipes_subprocess_client": PipesSubprocessClient()},
+)
+```
+
+</TabItem>
+<TabItem name="Report from the external code">
+
+From the external code, you can report to Dagster that an asset check has been performed via <PyObject module="dagster_pipes" object="PipesContext" method="report_asset_check" />. Note that `asset_key` in this case is required, and must match the asset key defined in `@asset_check`:
+
+```python file=/guides/dagster/dagster_pipes/subprocess/asset_check_external_code.py
+import pandas as pd
+from dagster_pipes import PipesContext, open_dagster_pipes
+
+
+def main():
+    orders_df = pd.DataFrame({"order_id": [1, 2], "item_id": [432, 878]})
+    # get the Dagster Pipes context
+    context = PipesContext.get()
+    # send structured metadata back to Dagster
+    context.report_asset_check(
+        asset_key="my_asset",
+        passed=orders_df[["item_id"]].notnull().all().bool(),
+        check_name="no_empty_order_check",
+    )
+
+
+if __name__ == "__main__":
+    # connect to Dagster Pipes
+    with open_dagster_pipes():
+        main()
+```
+
+</TabItem>
+</TabGroup>
+
+## Working with @multi_asset
```

Comment:
```suggestion
---

## Working with @multi_asset
```

Again, same ? here about using a decorator versus the plain English name (`multi-assets`)